File: commands/leaderboard.js
var axios = require("axios");
var Discord = require("discord.js");

module.exports = {
  name: "leaderboard",
  aliases: ["leaders", "lb"],
  description: "Current scoresaber global leaderboards.",
  usage: "b!leaderboard",
  async execute(msg, args) {
    msg.channel.startTyping();
    var config = {
      method: "get",
      url: "http://new.scoresaber.com/api/players/1",
      headers: {
        "Content-Type": "application/json",
        "User-Agent": "BeatSaberDiscordBot",
      },
    };

    axios(config)
      .then((res) => {
        var embedData = new Discord.MessageEmbed()
          .setColor("#309eff")
          .setTitle("🏆 **Top 10 Players Leaderboard.**")
          .setThumbnail("https://pbs.twimg.com/profile_images/1205366947543519232/ylj1LoJO_400x400.png")
          .setDescription(getTenPlayers(res.data.players))
          .setFooter(
            `Data fetched from ScoreSaber.`,
            "https://cdn.discordapp.com/attachments/478201257417244675/760182130352586802/unknown.png",
          );

        msg.channel.send(embedData);

        msg.channel.stopTyping();
      })
      .catch((res) => {
        console.log(err);
        msg.channel.send(
          "❌ There was an error trying to execute that command!",
        );
        msg.channel.stopTyping();
      });
  },
};

function getTenPlayers(input) {
  var data = [];

  for (i = 0; i < 10; i++) {
    var player = input[i];
    data.push(
      `\`${player.rank}.\` [${player.playerName}](https://new.scoresaber.com/u/${player.playerId}) | **PP:** ${player.pp} | **Country:** ${player.country} :flag_${player.country.toLowerCase()}:`,
    );
  }

  return data;
}

File: cogs/leaderboard.py
import discord, asyncio, random, time, datetime, json, aiohttp, requests, humanize, psutil, logging, platform
from discord.ext import commands
from discord.ext.commands.cooldowns import BucketType
from disputils import BotEmbedPaginator
from discord import Embed

log = logging.getLogger(__name__)

with open("config.json") as f:
    config = json.load(f)

class LeaderboardCommands(commands.Cog):
    """ Commands that use the ScoreSaber leaderboard """
    def __init__(self, bot):
        self.bot = bot
    
    @commands.cooldown(1, 5, BucketType.user)
    @commands.command()
    async def leaderboard(self, ctx):
        """Current scoresaber global leaderboard info"""
        async with ctx.typing():
            async with aiohttp.ClientSession() as session:
                async with session.get("https://new.scoresaber.com/api/players/1") as r:
                    result = await r.json()
            data = ''
            embed = []
            for i in range (50):
                player = result['players'][i]
                data+=(f"`#{player['rank']}.` **{player['playerName']}** | **PP:** {player['pp']} | **Country:** {player['country']} :flag_{player['country'].lower()}:\n")
                if i == 9 or i == 19 or i == 29 or i == 39 or i == 49:
                    e = Embed(title=f"🏆 Global Leaderboards",
                        description=data,
                        color=0x115599)
                    e.set_footer(text="Do b!profile name <user> for more info", icon_url="https://cdn.discordapp.com/attachments/753288806928482354/811991289645760542/rYiHR2pu_400x400.png")
                    embed.append(e)
                    data = ''
                    e = ''
        paginator = BotEmbedPaginator(ctx, embed)
        try:
            await paginator.run()
        except Exception:
            await ctx.send("**:no_entry: Error! Make sure I have the correct permissions to use this command. I may need `manage_messages` perms or `add_reactions` perms. If that doesn't fix the issue, contact my devs!**")

def setup(bot):
    bot.add_cog(LeaderboardCommands(bot))
